src/extractor/guidance/sliproad_handler.cpp

```diff
@@ -499,11 +501,22 @@ operator()(const NodeID /*nid*/, const EdgeID source_edge_id, Intersection inter
                 node_data_container
                     .GetAnnotation(node_based_graph.GetEdgeData(main_road.eid).annotation_data)
                     .name_id;
+            const auto main_road_name_empty =
+                main_road_name_id == EMPTY_NAMEID
+                    ? true
+                    : name_table.GetNameForID(main_road_name_id).empty();
```

Comment:
Here and below: you can simplify to

    const auto empty = name_id == EMPTY || GetName(name_id).empty()